core/src/main/java/com/netflix/conductor/service/WorkflowValidatorImpl.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.conductor.service;
+
+import com.netflix.conductor.common.metadata.workflow.WorkflowDef;
+import com.netflix.conductor.common.run.Workflow;
+import com.netflix.conductor.core.execution.ValidationException;
+import org.everit.json.schema.Schema;
+import org.everit.json.schema.loader.SchemaLoader;
+import org.json.JSONObject;
+
+import java.util.List;
+
+
+public class WorkflowValidatorImpl implements WorkflowValidator {
+	@Override
+	public void validate(WorkflowDef workflowDef, Workflow workflow) {
+		if (workflowDef.getInputDefinition().isEmpty()) {
+			return;
```

Comment:
what if the `workflow` has some input? shouldn't this be invalid? (as no input is expected)

Reply:
This line was used to avoid validation in case of old workflows that are not yet using the inputDefinition. However, after some tests, I realized that you need to explicitly define your object in the JSON schema with 
`"additionalProperties": false`, if you want to mark the input as invalid if any non-specified parameter is present inside your object. So, we could think to add automatically this property to the root object (___input___), but I think the best would be to give the freedom to the end-user to define its own workflow